From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 2 Nov 2011 19:42:29 +0000
Subject: python: Pass $PYTHON environment variable to tests.

If the user set PYTHON when configuring, this variable is not passed
through to the tests, so it is possible the tests will fail because
they are testing the wrong version of python.  By passing $PYTHON
through to the tests we ensure that we test against the same version
of python that we configured with.
---
 python/Makefile.am      |    5 +++--
 python/run-bindtests    |    4 ++--
 python/run-python-tests |    4 ++--
 3 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/python/Makefile.am b/python/Makefile.am
index 97fae83..f26ee71 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -1,5 +1,5 @@
 # libguestfs Python bindings
-# Copyright (C) 2009 Red Hat Inc.
+# Copyright (C) 2009-2011 Red Hat Inc.
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -48,7 +48,8 @@ libguestfsmod_la_LDFLAGS = -avoid-version -shared
 TESTS_ENVIRONMENT = \
 	LIBGUESTFS_PATH=$(top_builddir)/appliance \
 	PYTHONPATH=$(builddir):$(builddir)/.libs \
-	TMPDIR=$(top_builddir)
+	TMPDIR=$(top_builddir) \
+	PYTHON=$(PYTHON)
 
 TESTS = run-bindtests
 
diff --git a/python/run-bindtests b/python/run-bindtests
index 08a24e6..cf28bb6 100755
--- a/python/run-bindtests
+++ b/python/run-bindtests
@@ -1,6 +1,6 @@
 #!/bin/sh -
 # libguestfs Perl bindings
-# Copyright (C) 2009 Red Hat Inc.
+# Copyright (C) 2009-2011 Red Hat Inc.
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -18,5 +18,5 @@
 
 set -e
 
-python ${srcdir}/bindtests.py > bindtests.tmp
+$PYTHON ${srcdir}/bindtests.py > bindtests.tmp
 diff -u ${srcdir}/../bindtests bindtests.tmp
diff --git a/python/run-python-tests b/python/run-python-tests
index 487f8ab..f82dc10 100755
--- a/python/run-python-tests
+++ b/python/run-python-tests
@@ -1,6 +1,6 @@
 #!/bin/sh -
 # libguestfs Perl bindings
-# Copyright (C) 2009 Red Hat Inc.
+# Copyright (C) 2009-2011 Red Hat Inc.
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -19,5 +19,5 @@
 set -e
 
 for f in t/*.py; do
-  python $f
+  $PYTHON $f
 done
-- 
